File: back-end/pydantec_schemas/attendance.py
from pydantic import BaseModel
from datetime import datetime
from .student import Student
from .classes import Class


class AttendanceBase(BaseModel):
    pass


class AttendanceCreate(AttendanceBase):
    class_id: int
    student_id: int
    isPresent: bool


class Attendance(AttendanceBase):
    attendance_id: int
    created_at: datetime
    updated_at: datetime
    student: Student
    classes: Class

    class config:
        orm_mode: True

File: front-end/src/types/attendance.ts
import { StudentClass } from "./class";
import { Student } from "./students";
import { Subject } from "./subject";

export type Attendance = {
  attendance_id: number;
  created_at: string;
  updated_at: string;
  student: Student;
  classes: StudentClass;
  subject: Subject;
};

export type AttendanceForm = {
  class_id: number;
  student_id: number;
  isPresent: boolean;
};
